Favorites on main (#404) and option to exclude favorites from history

Geaz opened the pull request:
Hi!

I came across this great launcher yesterday and was just missing a favorites bar above the search bar (like mentioned in #404).

This pull request adds the option to show the favorites above the search bar. If the option is activated the favorites on the kiss bar (the one, if you touch the circle) disappears.
Furthermore I thought that it would be great to be able to exclude the favorites from the history. This is the second option I added to the settings menu.

This is my first contribution to an android/java application. So please be kind, if I missed something :)

Best Regards

licaon-kter commented:
While I don't care about a favorites bar above the search, I do like the "exclude favorite from history" option. :+1:

nmitsou commented:
I have just run this and I like what I see :+1: 

I haven't checked the code yet, I would suggest that the default value is false on PreferenceManager.getDefaultSharedPreferences(this).getBoolean("enable-favorites-bar", true)

Whats the difference between the favsIds and favBarIds ? Do we need both?

alexander255 commented on app/src/main/java/fr/neamar/kiss/SettingsActivity.java:
```diff
@@ -26,7 +26,7 @@
 
     // Those settings require the app to restart
-    final static private String requireRestartSettings = "theme enable-spellcheck force-portrait";
+    final static private String requireRestartSettings = "theme enable-spellcheck force-portrait enable-favorites-bar";
```
Is there a technical reason why `enable-favorites-bar` *must* require a restart? Showing/hiding the favorites bar can also be done in `onResume` and calling each time the launcher is resumed is probably no big burden either (and, if I interpret the code correctly) makes the launcher respond better to app icon changes from themes, upgrades, …

alexander255 commented on app/src/main/res/values-de/strings.xml:
```diff
@@ -115,6 +115,8 @@
     <string name="enable_sms">Eingehende Textnachrichten im Verlauf anzeigen</string>
     <string name="enable_phone">Eingehende Anrufe im Verlauf anzeigen</string>
     <string name="enable_app">Neu installierte Apps im Verlauf anzeigen</string>
+    <string name="enable_favorites_bar">Favoriten über der Suche anzeigen</string>
```
Nur eine Kleinigkeit, aber der "offizielle" Begriff ist in diesem Fall *Suchfeld* nicht *Suche* (s. `main_clear`)

alexander255 commented on app/src/main/res/values/strings.xml:
```diff
@@ -128,6 +128,8 @@
     <string name="enable_sms">Show incoming SMS in history</string>
     <string name="enable_phone">Show incoming calls in history</string>
     <string name="enable_app">Show newly installed apps in history</string>
+    <string name="enable_favorites_bar">Show favorites above search</string>
```
Same thing here: *search field*, not *search*

alexander255 commented:
Good looks good to me, just get rid of the unnecessary restart and I'm :+1:

Geaz commented:
Thanks for the review!
Already changed the following things:

- Strings changed @alexander255 
- Removed restart @alexander255 
- Changed default values of the two new settings @nmitsou 

@nmitsou mentioned the favBarIds. I am not happy with this either. Would it maybe better to move the favorites bar layout out to a seperate layout file and then to dynamiclly add this layout either to the main view or the kiss bar - depending on the current setting? This way we could remove redudancy.

nmitsou commented:
Good job @Geaz . 
In the mockup of #404, the icons in the fav bar have the same height as the ones in the history list.
To me, the mockup looks a bit nicer (visually). 
Should we increase the height of the fav bar?

PS: I think the fav list remains empty until the user clicks the green circle and then is populated

Geaz commented:
Here are screenshots with the current height and the same height as the icons in the history.
I like the bigger fav bar more, too.

**Current:**
![screenshot_20160716-161523](https://cloud.githubusercontent.com/assets/1579505/16895277/b3194ab2-4b71-11e6-92c4-c7b4da48a2cd.png)

**Same height as history icons:**
![screenshot_20160716-161453](https://cloud.githubusercontent.com/assets/1579505/16895276/b319235c-4b71-11e6-9178-b4e25972bc34.png)

nmitsou commented:
I would vote for the big ones. 
Should the bar be allways visible on minimalistic ui or only on click? I cannot decide what I like most

licaon-kter commented:
Toggle toggles toggles :-)

Geaz commented:
Changed the Icon size now and fixed a bug mentioned by @nmitsou with the empty fav bar on start.
Will now add a setting to hide the fav bar on minimalistic ui.

alexander255 commented:
Does it really make sense to have a toggle the display of favorite entries as part of the search results? Why not just always hide them whenever the favorites bar is visible on the main screen?
Better to be smart than configurable IMHO…

Neamar commented:
Hello, and thanks for your contribution!

That does sound great. I agree with @alexander255  that the second toggle can be removed :)

I'll need to have a deeper look at the source code, but for now I'm just going to set that up on my phone and use for a few day to get a better feeling. I think I'm going to love that so much more than the current implementation... :shipit:

Geaz commented:
Removing the toggle regarding "exclude favorites from history" would also prevent other users, which are using the favorites in the kiss bar, from enjoying this feature. I think there a users out there which are using the favorites in the kiss bar AND don't wanting the favorites in the history. What do you think?

licaon-kter commented:
Yes, I want that toggle.

Neamar commented:
Ok for the second toggle then. 
After a day of use, I'm wondering if the favorite bar should disappear once you start typing a query, making room for results.  It would make sense to me, what do you think?

nmitsou commented:
@neamar yes it makes sense
This is what is happening now (bug) :
On minimalistic ui with fav bar visible on click,  when I return to kiss and the search field contains text from the previous time the bar is not visible (but the history is)

Geaz commented:
I implemented the hiding of the favs bar during the search now.
Furthermore I merged the latest changes of your master branch.

Neamar commented:
So, everything seems fine... thanks! Just one last small issue yet, if loading takes too long, the "No favorites yet" toast will be displayed, I'll add the commit reference once I merge this.

Thanks again!

Neamar commented:
Commit reference:  1b5784d
